test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/CDEServersView.java

```diff
@@ -10,21 +10,30 @@
  ******************************************************************************/
 package org.jboss.tools.cdk.reddeer.server.ui;
 
+import org.eclipse.reddeer.common.logging.Logger;
+import org.eclipse.reddeer.eclipse.exception.EclipseLayerException;
 import org.eclipse.reddeer.eclipse.wst.server.ui.cnf.Server;
 import org.eclipse.reddeer.eclipse.wst.server.ui.cnf.ServersView2;
 
 public class CDEServersView extends ServersView2 {
 	
 	protected boolean cdk3 = false;
 	
+	private static final Logger log = Logger.getLogger(CDEServersView.class);
+	
 	public CDEServersView(boolean cdk3) {
 		this.cdk3 = cdk3;
 	}
 
 	@Override
 	public Server getServer(String name) {
-		return super.getServer(CDEServer.class, name);
+		for (Server server : getServers()){
+			if (server.isValid() && server.getLabel().getName().equals(name)){
+				return new CDEServer(server.getTreeItem(), this.cdk3);
+			}
+		}
+		log.info("Requested server '" + name + "' was not found on Servers view");
+		throw new EclipseLayerException("There is no server with name " + name);
```

Comment:
the only difference compared to the super impl I see is that you're not logging the available server. What do I miss?

Reply:
@odockal oh, I see, you need to be able to create the adapter with a different constructor that includes the flag cdk3. 
Why would you not override CDEServer#getAdapterConstructorArguments() to include the boolean?
Looking at CDEServer I'd think that the constructor with the treeitem should call the 2-argument constructor instead of super for good practice anyhow.
IMHO some sort of overriding of the differing part is preferable to duplication of the identical parts.